sprocs/assessment_instances_select_log.sql

```diff
@@ -24,6 +24,30 @@ AS $$
 BEGIN
     RETURN query
         WITH
+        ai_group_users AS (
+            SELECT gl.user_id
+            FROM
+                assessment_instances AS ai
+                JOIN group_logs AS gl ON (gl.group_id = ai.group_id)
+            WHERE
+                ai.id = ai_id
+                AND gl.action = 'join'
+        ),
+        user_page_view_logs AS (
+            SELECT pvl.*
+            FROM
+                page_view_logs AS pvl
+                JOIN assessment_instances AS ai ON (ai.id = pvl.assessment_instance_id)
+            WHERE
+                pvl.assessment_instance_id = ai_id
+                -- Only include events for the assessment's user or, in case of
+                -- group assessments, for events triggered by any user that at
+                -- some point was part of the group.
+                AND (
+                    (ai.user_id IS NOT NULL AND pvl.authn_user_id = ai.user_id)
```

Comment:
The `ai.user_id IS NOT NULL` is new here. I don't think it's strictly necessary, but it matches what we do on the next line for groups? Alternatively, these two lines could instead become the following, since `ai_group_users` should be an empty set for non-group assessments:
```sql
AND (pvl.authn_user_id = ai.user_id OR pvl.authn_user_id IN (SELECT * FROM ai_group_users)
```

Reply:
And why are we using `pvl.authn_user_id` rather than `pvl.user_id`? We should be using the effective user ID in almost all cases, I think?

Reply:
I agree with Matt, IS NOT NULL looks redundant. As for authn, I believe we don't want to include in the logs page views by the student if the instructor is viewing as effective student?

Reply:
We've used `authn_user_id` for as long as this query has been around - see https://github.com/PrairieLearn/PrairieLearn/blame/ef36c97de71cde0a6b05f3dc6eb1a12646bee03e/pages/instructorAssessmentInstance/instructorAssessmentInstance.sql (this is before it made the leap to a sproc). I imagine there's value in differentiating between "instructor impersonating student and viewing variant" and "student viewing variant"?

Reply:
I dropped the `IS NOT NULL` bits in 4d37fde. I'm going to stick with `authn_user_id` unless we have a really compelling reason to change it.

Reply:
Fair enough on `authn_user_id`. At some point we should make some firm rules about when an instructor emulating a student should show up in logs.